lib/plugins/aws/invokeLocal/resolvers/index.js

```diff
@@ -0,0 +1,160 @@
+const aws = require('aws-sdk');
+const fs = require('fs');
+
+const handlers = {
+    'AWS::S3::Bucket': require('./s3'),
+    'AWS::DynamoDB::Table': require('./dynamodb'),
+    'AWS::SNS::Topic': require('./sns'),
+    'AWS::SQS::Queue': require('./sqs'),
+    'AWS::Lambda::Function': require('./lambda')
+};
+
+let cfrServerless = null;
+let cfResources = null;
+let cfExports = null;
+let serviceConfig = null;
+let cache = null;
+const cacheFilePath = '.serverless/invoke.cache.json';
+
+async function getCFResources(serverless) {
+    if(cfrServerless === serverless) {
+        return cfResources;
+    }
+
+    const cf = new aws.CloudFormation(serviceConfig);
+
+    const resources = await cf.describeStackResources({
+        StackName: `${serverless.service.service}-${serverless.service.provider.stage}`
+    }).promise();
+
+    return resources.StackResources;
+}
+
+module.exports = async (name, environment, serverless, credentialEnvVars) => {
+    if(!cache) {
+        if(fs.existsSync(cacheFilePath)) {
+            try {
+                cache = JSON.parse(fs.readFileSync(cacheFilePath).toString('utf-8'));
+            } catch (err) {
+                cache = {};
+            }
+        } else {
+            cache = {};
+        }
+    }
+    const value = environment[name];
+    const accessKey = credentialEnvVars.AWS_ACCESS_KEY_ID;
+    const secret = credentialEnvVars.AWS_SECRET_ACCESS_KEY;
+    serviceConfig = {
+        region: (credentialEnvVars.AWS_REGION)? credentialEnvVars.AWS_REGION : serverless.service.provider.region,
+        credentials: (accessKey)? new AWS.Credentials(accessKey, secret) : undefined
+    };
+
+    if(typeof value != 'object') {
+        return;
+    }
+
+    const keys = Object.keys(value);
+    if(keys.length != 1) {
+        // Not sure how to handle this where an environment variable has multiple opperators
+        return;
+    }
+    const action = keys[0];
+
+    let resourceName = null;
+    let importName = null;
+    switch(action) {
+        case 'Ref':
+            resourceName = value[action];
+            break;
+        case 'Fn::GetAtt':
+            resourceName = value[action][0];
+            break;
+        case 'Fn::ImportValue':
+            importName = value[action];
+            break;
+    }
+
+    if(cache[action] && cache[action][resourceName]) {
+        switch(action) {
+            case 'Ref':
+            case 'Fn::ImportValue':
+                environment[name] = cache[action][resourceName];
+                return;
+            case 'Fn::GetAtt':
+                const val = cache[action][resourceName][value[action][1]];
+                if(val) {
+                    environment[name] = val;
+                    return;
+                }
+                break;
+        }
+    }
+
+    if(resourceName) {
+        let resource = serverless.service.resources.Resources[resourceName];
+        if(!resource && resourceName.endsWith('LambdaFunction')) {
```

Comment:
This may work ok only for common cases, but won't where user decides on custom function name, or in some other custom way configures resource.

To ensure it's bulletproof, I think any resource resolution should be made out of generated cloudformation template, which can be find at:

```javascript
this.serverless.service.provider.compiledCloudFormationTemplate.Resources
```

Reply:
Does serverless framework compile cloudformation when doing a cold debug?  I tend not to see any '.serverless' directory made when invoking locally without having one already existing

Reply:
No, it wasn't so far, as there was no need to inspect generated CF template for local invocation, but this update is about to change it (in my understanding we cannot reliably implement it, without such inspection)